test/puppetlabs/puppetdb/scf/storage_test.clj

```diff
@@ -686,7 +687,28 @@
                 ["foo-2" "1.2.3" "apt"]
                 ["foo-3" "1.2.3" "apt"]
                 ["foo-4" "1.2.3" "apt"]]
-               (package-seq certname)))))))
+               (package-seq certname))))
+
+      (testing "Remove all packages, then GC"
+        (update-facts!
+         {:certname certname
+          :values facts
+          :timestamp (-> 1 days ago)
+          :environment "DEV"
+          :producer_timestamp (-> 1 days ago)
+          :producer producer
+          :package_inventory [["foo-1" "1.2.3" "apt"]]})
+
+        (is (= [["foo-1" "1.2.3" "apt"]]
+               (package-seq certname)))
+
+        (delete-unassociated-packages!)
+
+        (is (= 1
+               (-> ["SELECT count(*) as c FROM packages"]
+                   query-to-vec
+                   first
+                   :c)))))))
```

Comment:
I don't really understand what this is testing. It seems like nothing is being removed?